core/src/main/java/org/arquillian/smart/testing/configuration/ConfigurationLoader.java

```diff
@@ -22,36 +22,60 @@
 
     public static final String SMART_TESTING_YML = "smart-testing.yml";
     public static final String SMART_TESTING_YAML = "smart-testing.yaml";
+    public static final String SMART_TESTING_CONFIG = "smart.testing.config";
+
     private static final Logger logger = Log.getLogger();
 
     public static Configuration load() {
         return load(Paths.get("").toAbsolutePath().toFile());
     }
 
     public static Configuration load(File projectDir) {
+        Map<String, Object> yamlConfiguration = new HashMap<>(0);
+
+        final String customConfigFile = System.getProperty(SMART_TESTING_CONFIG);
+        if (customConfigFile != null) {
+            yamlConfiguration = getConfigurationFromCustomConfigFile(customConfigFile, yamlConfiguration);
```

Comment:
Providing the `yamlConfiguration` instance as a method parameter and then getting the very same instance as a return value of the method looks weird. Use one of the solutions, not both at once. The method name should be changed to reflect what it really does-in other words use:
either:
`loadConfigParametersFromCustomConfigFile(customConfigFile, yamlConfiguration);`
or:
`yamlConfiguration = getConfigParametersFromCustomConfigFile(customConfigFile);`